src/vl53l1_class.cpp

```diff
@@ -3893,8 +3893,16 @@ VL53L1_Error VL53L1::VL53L1_GetTickCount(uint32_t *ptick_count_ms)
 
 VL53L1_Error VL53L1::VL53L1_WaitUs(VL53L1_Dev_t *pdev, int32_t wait_us)
 {
-  (void)pdev;
-  delay(wait_us / 1000);
+  /* Currently, the largest value that will produce an accurate delay is 16383.
+   * This could change in future Arduino releases.
+   * https://www.arduino.cc/reference/en/language/functions/time/delaymicroseconds/
+   */
+  if (wait_us > 16383) {
+    VL53L1_WaitMs(pdev, wait_us / 1000);
+    delayMicroseconds(wait_us % 1000);
+  } else {
+    delayMicroseconds(wait_us);
+  }
```

Comment:
Here I would simply do:
```suggestion
  VL53L1_WaitMs(pdev, wait_us / 1000);
  delayMicroseconds(wait_us % 1000);
```

Any feedback @cparata ?

Reply:
I agree, but the indentation must be smaller.
```suggestion
  VL53L1_WaitMs(pdev, wait_us / 1000);
  delayMicroseconds(wait_us % 1000);
```

Reply:
Right. 😉